.github/workflows/build.yml

```diff
@@ -30,7 +30,28 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  tidy:
+    runs-on: ubuntu-latest
+    name: Tidy
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.ADD_TO_PROJECT_PAT }}
+      - name: Setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{env.GO_VERSION}}
+      - run: |
+          make tidy-workspace
+      - uses: EndBug/add-and-commit@v9
```

Comment:
will this fail the workflow if `make tidy-workspace` introduce no changes?

Reply:
It will not fail @pysel

Reply:
Can you explain more clearly the meaning of `Is it possible not to run tidy step when a PR is merged?`

Reply:
this workflow runs on push to main (in other words, when a PR is merged). So I am worried that after some PR gets merged, this workflow will commit changes introduced by `tidy-workspace` directly to main (if I understand correctly), if any

Reply:
Ok, I understand. My opinion is:
1. Maybe we will never push a commit directly to the main branch, right? So the case of pushing directly to the main can be eliminated.
2. In case of merging PR to main, CI will be run in PR first so it will create a commit on the PR branch so I see no problem here.